### PR TITLE
Announce a breaking change of added `dispose()`.

### PR DESCRIPTION
**Pull request by polina-c**
Discussion: https://docs.google.com/document/d/1ec5djPP14b1u93O9fHrJJf3gbONay1nV4sBV2-XXOIw/edit?resourcekey=0-CFEu4HWAW8yP0knhoVfKxQ

This announce is added as already released, because we are adding `dispose` already long time and some of the changes may be already released: https://github.com/flutter/devtools/issues/3951

Tech writers, please do not merge this before tech sign off. @goderbauer , can you review please?

### CODE REVIEW

File: src/release/breaking-changes/dispose.md
Comment (atsansone) [inline src/release/breaking-changes/dispose.md:11]: **chore (if-minor):** Use [semantic breaks](https://sembr.org/).
```suggestion
Missed 'dispose()' are added for some disposable objects.
If some other code also invokes 'dispose()' for the objects,
and the object is protected from double disposal,
the second 'dispose()' fails with the following error message:
```

File: src/release/breaking-changes/dispose.md
Comment (atsansone) [inline src/release/breaking-changes/dispose.md:21]: **chore (if-minor):** Use [semantic breaks](https://sembr.org/). Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure). Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
This rule was broken in some places in Flutter Framework:
owners were not disposing the disposable objects.
To fix the issue, `dispose()` was added.
If there is another `dispose()`, and the object is protected from double disposal,
the added `dispose()` can cause failures in debug mode.
```

File: src/release/breaking-changes/dispose.md
Comment (atsansone) [inline src/release/breaking-changes/dispose.md:27]: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
````suggestion
If the following error was raised for your code, 
update the code to call `dispose()' only in cases when your code created the object.

```nocode
Once you have called dispose() on a <class name>, it can no longer be used.
```
````

**Review by atsansone — APPROVED** [review]
@polina-c : A few suggestions which you can accept or not. LGTM % your review of those suggestions.

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [inline src/release/breaking-changes/dispose.md:4]:
```suggestion
  'dispose()' might fail because of double disposal.
```

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [reply src/release/breaking-changes/dispose.md:11]: Please fix the grammar:
Missed 'dispose()' are added for some disposable objects.
If some other code also invokes 'dispose()' for the objects,

=>
Missing calls to `dispose()` were added for some disposable objects.
If `dispose()` is called elsewhere on the same object

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [inline src/release/breaking-changes/dispose.md:17]: Is this a rule or a convention? If a convention:

```suggestion
The convention is that the owner of an object should dispose of it.
```

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [reply src/release/breaking-changes/dispose.md:21]: Please fix the grammar:

=>
This convention was broken in some places:
owners were not disposing the disposable objects.
The issue was fixed by adding a call to `dispose()`.
However, if the object is protected from double disposal,
this can cause failures when running in debug mode
and `dispose()` is called elsewhere on the object.

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [reply src/release/breaking-changes/dispose.md:27]: Please fix the grammar:
If you encounter the following error, then remove
any additional calls in your code that dispose of the object:

```nocode
Once you have called dispose() on a <class name>, it can no longer be used.
```

File: src/release/breaking-changes/dispose.md
Comment (sfshaza2) [inline src/release/breaking-changes/dispose.md:47]: It seems like you don't really need this first sentence?
If so, delete. That second sentence is super useful.

```suggestion
If the error occurs in Flutter code, `dispose()` was
called incorrectly the first time.

You can locate an incorrect call to `dispose()` by
temporary calling `print(StackTrace.current)`
in the body of the failed `dispose` method.
```

File: src/release/breaking-changes/index.md
Comment (sfshaza2) [inline src/release/breaking-changes/index.md:53]:
```suggestion
* [Add missing calls to `dispose()` for some disposable objects in Flutter][]
```

File: src/release/breaking-changes/index.md
Comment (sfshaza2) [inline src/release/breaking-changes/index.md:63]:
```suggestion
[Add missing calls to `dispose()` for some disposable objects in Flutter]: {{site.url}}/release/breaking-changes/dispose
```

**Review by sfshaza2 — COMMENTED** [review]
A few grammatical improvements.

File: src/release/breaking-changes/dispose.md
Comment (polina-c) [reply src/release/breaking-changes/dispose.md:28]: Shams, in most cases, the code should not be removed, but condition should be added. So, it should be fixed, not removed.

File: src/release/breaking-changes/dispose.md
Comment (polina-c) [reply src/release/breaking-changes/dispose.md]: print(StackTrace.current) makes sense if the error is for Flutter code, otherwise it is not needed.
I updated text to made it clearer.

File: src/release/breaking-changes/dispose.md
Comment (kenzieschmoll) [inline src/release/breaking-changes/dispose.md:9]: nit: Missing calls to 'dispose()'

File: src/release/breaking-changes/index.md
Comment (kenzieschmoll) [inline src/release/breaking-changes/index.md:41]: was this intentionally deleted?

**Review by kenzieschmoll — APPROVED** [review]
LGTM, but @goderbauer should review before merge

File: src/release/breaking-changes/index.md
Comment (polina-c) [reply src/release/breaking-changes/index.md]: returned. thanks

File: src/release/breaking-changes/dispose.md
Comment (goderbauer) [inline src/release/breaking-changes/dispose.md:9]: For added context it would be nice to list somewhere in this doc examples of where we added previously missing disposes, e.g. "The FooController owned by the BarWidget was missing a dispose" or something like that.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**
